liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -780,6 +823,29 @@ private void runMojo(String groupId, String artifactId, String defaultVersion, S
                 executionEnvironment(project, session, pluginManager));
     }
 
+    private void runBoostMojo(String goal, boolean rebuildProject)
+            throws MojoExecutionException, ProjectBuildingException {
+
+        Plugin boostPlugin = project.getPlugin(Plugin.constructKey("boost", "boost-maven-plugin"));
+
+        MavenProject boostProject = this.project;
+        MavenSession boostSession = this.session;
+
+        if (rebuildProject) {
+            // Reload pom
+            File pomFile = new File(project.getFile().getAbsolutePath());
+            ProjectBuildingResult build = mavenProjectBuilder.build(pomFile,
+                    session.getProjectBuildingRequest().setResolveDependencies(true));
+            boostProject = build.getProject();
```

Comment:
So how/why this is different from the non-Boost path using `DevMojo.loadProject(File)`  to reload the POM?    Is it mainly because we're trying to fluff up objects needed by the mojo executor?  Some other reason?

Reply:
Right - we are passing an actual maven project object to Boost via the mojo executor, so we need that project to have all of its dependency information resolved (i.e. versions when boms are being used). DevMojo.loadProject(File) does not do that, which is ok for that path because it uses an artifact resolver later on to resolve each individual dependency. For our case, we need to use the MavenProjectBuilder which builds a new maven project object with dependency versions correctly resolved.